src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/MissingLeadingAsteriskCheck.java

```diff
@@ -0,0 +1,139 @@
+
+import java.util.regex.Pattern;
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.AbstractCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtil;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks if the javadoc has leading asterisk on each line or not.
+ * </p>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name="MissingLeadingAsterisk"/&gt;
+ * </pre>
+ * <p>
+ * Example:
+ * </p>
+ * <pre>
+ * &#47;** // violation, javadoc has lines without leading asterisk.
+ *  * Some description here.
+ *    Another line of description.
+ *  *&#47;
+ * public class Test {
+ *   &#47;** // violation, javadoc has lines without leading asterisk.
+ *    * Some description here.
+ *    * Another line of description.
+ *      Another line of description.
+ *    *&#47;
+ *   public void test() {}
+ *
+ *   &#47;** Some description here. *&#47; // OK
+ *   public void test1() {}
+ *
+ *   &#47;** // OK
+ *    * Some description here.
+ *    * Another line of description.
+ *    *&#47;
+ *   public void test2() {}
+ *
+ *   &#47;** // violation, javadoc has lines without leading asterisk.
+ *    * Some description here.
+ *
+ *    * Another line of description.
+ *    *&#47;
+ *   public void test3() {}
+ * }
+ * </pre>
+ *
+ * @since 8.32
+ */
+@StatelessCheck
+public class MissingLeadingAsteriskCheck extends AbstractCheck {
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_MISSING_ASTERISK = "javadoc.missing.asterisk";
+
+    /**
+     * The new line character that separates each line of javadoc.
+     */
+    private static final Pattern MATCH_COMMENT_SEPARATOR = CommonUtil.createPattern("\r\n?|\n");
+
+    @Override
+    public int[] getDefaultTokens() {
+        return new int[] {
+            TokenTypes.BLOCK_COMMENT_BEGIN,
+        };
+    }
+
+    @Override
+    public int[] getAcceptableTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public int[] getRequiredTokens() {
+        return getDefaultTokens();
+    }
+
+    @Override
+    public boolean isCommentNodesRequired() {
+        return true;
+    }
+
+    @Override
+    public void visitToken(DetailAST ast) {
+        final String commentContent = JavadocUtil.getBlockCommentContent(ast);
+        if (JavadocUtil.isJavadocComment(commentContent)) {
+            final String[] commentLines = MATCH_COMMENT_SEPARATOR.split(commentContent);
+
+            for (int i = 0; i < commentLines.length; i++) {
+                final String commentLine = commentLines[i];
+                final boolean isLast = i + 1 == commentLines.length;
+                if (!isValidLine(commentLine, isLast)) {
+                    log(ast, MSG_MISSING_ASTERISK);
```

Comment:
can we do ` for (int i = 0; i < commentLines.length - 1; i++) {` ?
and validate last line is special way after FOR-loop, logic in `isValidLine` is very complicated.

Reply:
Sure, we can iterate till `commentLines.length - 1` and the check the last line after the for loop ends. Is this okay ?

Reply:
Done